### DISCUSSION TITLE
preparing a transversely polarized Ising configuration for relaxation

### DISCUSSION THREAD

**Discussion by dcyang** [netket/netket#1730] [Q&A] [answered]
Hi, 
I have a question related to #1549 (TDVP) and #1672 (initial sample).
I've been trying to reproduce the results of Schmitt and Heyl ([2018](https://scipost.org/10.21468/SciPostPhys.4.2.013), [2020](https://journals.aps.org/prl/abstract/10.1103/PhysRevLett.125.100503)) with the transverse field Ising model.
One of their main results show the relaxation of a fully $x$-polarized spin configuration to a small transverse field.

My attempt at it was to use 2 Hamiltonians (`nk.operator.IsingJax`): first for equilibrating the system with a strong field (say $h = 99$) using `nk.driver.VMC`, and then pass the `MCState` to the second Hamiltonian with a weak field (say $h = 0.1$) using `nkx.driver.TDVPSchmitt`.

```python
L = 4
ham1 = nk.operator.IsingJax(nk.hilbert.Spin(1/2, L*L), nk.graph.Square(L), h=99, J=-0.25)
# ... running VMC to equilibrate vstate1 ...
ham2 = nk.operator.IsingJax(nk.hilbert.Spin(1/2, L*L), nk.graph.Square(L), h=0.1, J=-0.25)
time_ev = nkx.driver.TDVPSchmitt(operator=ham2, variational_state=vstate1,
             integrator=nkx.dynamics.Euler(dt=0.01), holomorphic=True)
```

But it looks like the effect of the initial strong field remains in the `MCState` as we go to the relaxation stage, so there is an overall drift in the total spin in the $x$ direction (sum of `nk.operator.spin.sigmax`). Below are $S_x$ relaxation curves of a 4x4 system to $h = 2 h_c$ on `nk.models.RBM` which were first equilibrated with $8 h_c$, $16 h_c$, $32 h_c$ transverse fields. They show _some_ oscillation but also drifts overall. So it seems to work partially ...

<img src="https://github.com/netket/netket/assets/3697637/25c15884-18d4-4234-9704-580f09cc9289" width="400"/>

Q: Is there a way to "manually" prepare a fully $x$-polarized configuration, ie. without the field effects from the equilibration stage?
(or will I have to write a new `nk.hilbert` object?)

Thanks in advance.

**Comment by PhilipVinc** [comment, marked as answer]
Well, to prepare a fully polarised state I would set J=0.0.

To do it a bit more cleanly and efficiently, I would use the [state learning fidelity minimisation example from netket_fidelity](https://github.com/netket/netket_fidelity/blob/main/examples/state_learning/state_learning.py). 
Just set J=0.0 in the example and you will be learning that state.

While this works starting from the state obtained through ED, it's trivial to write a 'variational ansatz' that prepares the fully polarised state (it's simply a constant function) that can be used for arbitrary sizes.

**Reply by PhilipVinc** [reply]
Furthermore, in my experience TDVP won't work with such large tilmestep.
You will need at most 1e-3 and possibly smaller.
If I remember correctly, Markus was using `Heun` in the PRL with adaptive time stepping (though adaptive time stepping is not very stable in my experience, so I'd suggest disabling it at the beginning).

Also, be sure to scan different values of [`rcond_smooth`](https://netket.readthedocs.io/en/v3.11.1/api/_generated/experimental/dynamics/netket.experimental.driver.TDVPSchmitt.html) to find the one that works well.

**Reply by dcyang** [reply]
Took me a while to try all the things you said. The drift persisted even after setting $J=0$ to prepare a `nk.VMC` state or learning it with netket_fidelity, so I concluded that the initial config is only a partial cause of the issue.

I have since started experimenting with different `nk.models` and saw that the effect is much weaker for some models than others.
Thanks @PhilipVinc for pointing me in the right direction!
